source/core/slang-performance-profiler.cpp

```diff
@@ -39,11 +41,53 @@ namespace Slang
             for (auto func : data)
             {
                 out << func.key << ": \t";
-                out << func.value.invocationCount << "\t" << func.value.duration.count() / 1000000.0f << "\n";
+                auto milliseconds = std::chrono::duration_cast< std::chrono::milliseconds >(func.value.duration);
+                out << func.value.invocationCount << "\t" << milliseconds.count() << "ms\n";
             }
         }
     };
 
+    class SerialPerformaceProfilerImpl : public SerialPerformaceProfiler
+    {
+    public:
+       void accumulateResults(StringBuilder& out, PerformanceProfiler* profile)
+       {
+           PerformanceProfilerImpl* profilerImpl = static_cast<PerformanceProfilerImpl*>(profile);
+            for (auto func : profilerImpl->data)
+            {
+                auto milliseconds = std::chrono::duration_cast< std::chrono::milliseconds >(func.value.duration);
+                long ms = milliseconds.count();
+
+                m_mutex.lock();
+                auto entry = m_compileTimeProfiler.find(func.key);
+                if (entry == m_compileTimeProfiler.end())
+                {
+                    m_compileTimeProfiler.emplace(std::make_pair(func.key, 0l));
+                    entry = m_compileTimeProfiler.find(func.key);
+                }
+                entry->second += ms;
+                m_mutex.unlock();
+            }
+
+            for (auto entry : m_compileTimeProfiler)
```

Comment:
This for-loop will need to be inside of a lock protection.
Because m_compileTimeProfiler can be modified while reading it.

Reply:
done.